src/main/MainScreenController.java

```diff
@@ -449,28 +458,154 @@ public static PlayerFish getPlayerFish() {
     return playerFish;
   }
 
-	public static EndBoss getEndBoss() {
-		return endBoss;
-	}
+  /**
+   * {@inheritDoc} Overrides the Handle method in AnimationTimer to contain the
+   * game loop.
+   * 
+   * @return the new AnimationTimer.
+   */
+  public static AnimationTimer makeAnimationTimer(GraphicsContext gc) {
+
+    return new AnimationTimer() {
+      public void handle(long currentNTime) {
 
-	public static void setEndBoss(EndBoss endBoss) {
-		MainScreenController.endBoss = endBoss;
-	}
+        BoundingBox endBossbb = endBoss.getSprite().getBoundingBox();
 
-	public static Lance getLance() {
-		return lance;
-	}
+        BoundingBox lancebb = lance.getSprite().getBoundingBox();
 
-	public static void setLance(Lance lance) {
-		MainScreenController.lance = lance;
-	}
+        Sprite pfSprite = playerFish.getSprite();
+        BoundingBox pfbb = pfSprite.getBoundingBox();
 
-	public static boolean isBossMode() {
-		return bossMode;
-	}
+        if (playerHasWon()) {
+          this.stop();
+          Game.switchScreen("FXML/WinningScreen.fxml");
+          Game.getMediaPlayer().stop();
+          Game.getLogger().logSwitchScreen("WinningScreen");
+
+          setBossMode(false);
+          playerFish.setHasLance(false);
+
+          endBossbb.setX(-2000);
+          endBossbb.setY(-2000);
+        }
+
+        if (pfbb.getWidth() > 60) {
+          if (endBossbb.getX() == -2000) {
+            endBossbb.setX(0);
+            endBossbb.setY(0);
+          }
+          if ((lancebb.getX() == -2000) && !playerFish.hasLance()) {
+            lancebb.setX(0);
+            BoundingBox screenBox = getScreenbox();
+            lancebb.setY(screenBox.getHeight() / 4 * 3);
+          }
+          setBossMode(true);
+        }
 
-	public static void setBossMode(boolean bossMode) {
-		MainScreenController.bossMode = bossMode;
-	}
+        if (currScore > 500 && !bomb1) {
+          playerFish.getBombs().add(FishBomb.createFishBomb(playerFish));
```

Comment:
But now when I run inCode, it shows MainScreenController is a Schizophrenic Class now.
< review